integration/docker/test_basic.py

```diff
@@ -170,8 +170,10 @@ def test_recurring_snapshot_live_update_retain(clients):  # NOQA
 
     time.sleep(15)
 
-    retained = filter(lambda s: s["removed"] is False, volume.snapshotList())
+    snapshots = volume.snapshotList()
+    retained = filter(lambda s: s["removed"] is False, snapshots)
     assert len(retained) == 3
+    assert 3 <= len(snapshots) <= 5
```

Comment:
This is not sufficient because if we didn't do purge, it will still be 5.

Since purge is done for the last two snapshots and none of them will have the HEAD as the parent, and here none of them will be the parent of another node, it should be purged and there should be only 3 snapshots shown. 

Do you see the problem that snapshots is not 3 but 4 or 5? If so, that worth investigation.